File: frontend/src/services/roles.ts
import { apiFetch } from "./api";

import type { Role } from "../types/roles";

export async function getRoles(): Promise<Role[]> {
  return apiFetch<Role[]>("/roles/");
}

export async function getHighRiskRoles(): Promise<Role[]> {
  return apiFetch<Role[]>("/roles/high-risk");
}

File: backend/app/services/roles.py
from typing import List, Dict, Any

from app.data.roles import ROLES


def get_all_roles() -> List[Dict[str, Any]]:
    """
    Return all enterprise IAM roles.
    """

    return ROLES


def get_high_risk_roles() -> List[Dict[str, Any]]:
    """
    Return roles classified as high risk.
    """

    return [
        role
        for role in ROLES
        if role["riskLevel"].lower() == "high"
    ]